### PR TITLE
Add repoonly option for otherpkgs to only generate repo for pkgdir and otherpkgdir

### PR DESCRIPTION
**Pull request by zet809**
### The PR is to add option `--repoonly` for otherpkgs script to generate repo based on pkgdir and otherpkgdir

_##Feature description##_
Only support rhels right now.

### The UT result
```
root@c910f03c05k10:/install/postscripts# updatenode c910f03c05k02 -P "otherpkgs --repoonly"
c910f03c05k02: Warning: the ECDSA host key for 'c910f03c05k02' differs from the key for the IP address '10.3.5.2'
c910f03c05k02: Offending key for IP in /root/.ssh/known_hosts:189
c910f03c05k02: Matching host key in /root/.ssh/known_hosts:341

c910f03c05k02: vpdupdate is not supported on the pSeries KVM Guest platform
c910f03c05k02: updating VPD database
c910f03c05k02: trying to download postscripts...
c910f03c05k02: postscripts downloaded successfully
c910f03c05k02: trying to get mypostscript from 10.3.5.10...
c910f03c05k02: Running postscript: otherpkgs
c910f03c05k02: There are no enabled repos.
c910f03c05k02:  Run "yum repolist all" to see the repos you have.
c910f03c05k02:  To enable Red Hat Subscription Management repositories:
c910f03c05k02:      subscription-manager repos --enable <repo>
c910f03c05k02:  To enable custom repositories:
c910f03c05k02:      yum-config-manager --enable <repo>
c910f03c05k02: pkgsarray: python2-matplotlib python2-scikit-image, 2
c910f03c05k02: yum: 1, apt: 0, zypper: 0
c910f03c05k02: Loaded plugins: product-id, search-disabled-repos, subscription-manager
c910f03c05k02: This system is not registered with an entitlement server. You can use subscription-manager to register.
c910f03c05k02: Cleaning repos: xCAT-rhels7.5-alternate-path0 xCAT-rhels7.5-alternate-path1
c910f03c05k02:               : xCAT-rhels7.5-alternate-path2 xCAT-rhels7.5-alternate-path3
c910f03c05k02:               : xCAT-rhels7.5-alternate-path4 xCAT-rhels7.5-alternate-path5
c910f03c05k02:               : xcat-otherpkgs0
c910f03c05k02: Cleaning up everything
c910f03c05k02: Maybe you want: rm -rf /var/cache/yum, to also free up space taken by orphaned data from disabled or removed repos
c910f03c05k02: Loaded plugins: product-id, search-disabled-repos, subscription-manager
c910f03c05k02: This system is not registered with an entitlement server. You can use subscription-manager to register.
c910f03c05k02: Cleaning repos: xCAT-rhels7.5-alternate-path0 xCAT-rhels7.5-alternate-path1
c910f03c05k02:               : xCAT-rhels7.5-alternate-path2 xCAT-rhels7.5-alternate-path3
c910f03c05k02:               : xCAT-rhels7.5-alternate-path4 xCAT-rhels7.5-alternate-path5
c910f03c05k02:               : xcat-otherpkgs0
c910f03c05k02: Cleaning up everything
c910f03c05k02: Maybe you want: rm -rf /var/cache/yum, to also free up space taken by orphaned data from disabled or removed repos
c910f03c05k02: otherpkgs: repoonly set, so ignore pkg installation ...
c910f03c05k02: postscript: otherpkgs exited with code 0
c910f03c05k02: Running of postscripts has completed.
```

### CODE REVIEW

File: xCAT/postscripts/otherpkgs [hunk 1]
```diff
@@ -213,7 +213,20 @@ apt_get_update_if_repos_changed()
...
+argnum=$#
+args=$@
+if [ $argnum -eq 1 ]; then
+    if ( pmatch "$args" "--repoonly" ); then
+        repoonly=1
+    else
+        echo "$(basename $0): option $args is not supported"
+        exit 0
+    fi
+else
```
Comment (robin2008) [inline xCAT/postscripts/otherpkgs]: if the script go with no args, here will exit with error. It is not what we want.

File: xCAT/postscripts/otherpkgs [hunk 2]
```diff
@@ -213,7 +213,20 @@ apt_get_update_if_repos_changed()
...
-
+repoonly=0
+argnum=$#
+args=$@
+if [ $argnum -eq 1 ]; then
+    if ( pmatch "$args" "--repoonly" ); then
+        repoonly=1
+    else
+        echo "$(basename $0): option $args is not supported"
+        exit 0
```
Comment (robin2008) [inline xCAT/postscripts/otherpkgs]: If error,  need to return non-zero.

File: xCAT/postscripts/otherpkgs [hunk 3]
```diff
@@ -231,7 +244,12 @@ if [ -z "$UPDATENODE" ] || [ $UPDATENODE -ne 1 ]; then
   fi
 fi
 
-if [ -z "$OTHERPKGS_INDEX" ]; then
+if ! pmatch $OSVER "rhel*" && [ "$repoonly" -eq 1 ]; then
+    echo "$0: the option \"repoonly\" only support rhel right now"
+    exit 0
+fi
+
+if [ -z "$OTHERPKGS_INDEX" ] && [ "$repoonly" -ne 1 ]; then
```
Comment (robin2008) [inline xCAT/postscripts/otherpkgs]: I would suggest that we have a separate if-clause for  "$repoonly", so we can print more clear message for it.

**Comment by zet809** [comment]
hi, @robin2008 , updated, pls review again.

**Pull request merged**